db/rdbms/src/main/java/io/camunda/db/rdbms/write/service/FlowNodeInstanceWriter.java

```diff
@@ -36,13 +35,26 @@ public void create(final FlowNodeInstanceDbModel flowNode) {
             flowNode));
   }
 
-  public void end(final long flowNodeKey, final FlowNodeState state, final OffsetDateTime endDate) {
-    final var dto = new EndFlowNodeDto(flowNodeKey, state, endDate);
-    executionQueue.executeInQueue(
-        new QueueItem(
-            ContextType.FLOW_NODE,
-            flowNodeKey,
-            "io.camunda.db.rdbms.sql.FlowNodeInstanceMapper.updateStateAndEndDate",
-            dto));
+  public void finish(final long key, final FlowNodeState state, final OffsetDateTime endDate) {
+    final boolean wasMerged = mergeToQueue(key, b -> b.state(state).endDate(endDate));
+
+    if (!wasMerged) {
+      final var dto = new EndFlowNodeDto(key, state, endDate);
+      executionQueue.executeInQueue(
+          new QueueItem(
+              ContextType.FLOW_NODE,
+              key,
+              "io.camunda.db.rdbms.sql.FlowNodeInstanceMapper.updateStateAndEndDate",
+              dto));
+    }
+  }
+
+  private boolean mergeToQueue(
+      final long key,
+      final Function<FlowNodeInstanceDbModelBuilder, FlowNodeInstanceDbModelBuilder>
+          mergeFunction) {
+    return executionQueue.tryMergeWithExistingQueueItem(
+        new UpsertMerger<>(
+            ContextType.FLOW_NODE, key, FlowNodeInstanceDbModel.class, mergeFunction));
```

Comment:
💭 A bit late but: Could these methods be moved to the `ExecutionQueue`? I think it would be nicer if the EQ would decide if it merges or just adds the item, instead of the writers. It also seems a bit repetitive this way.